### PR TITLE
New module - na_ontap_ldap_client

### PR DESCRIPTION
**Pull request by zeten30**
##### SUMMARY
Adding new storage/netapp module -  na_ontap_ldap_client - to manage[ ONTAP ldap client configuration](https://docs.netapp.com/ontap-9/index.jsp?topic=%2Fcom.netapp.doc.exp-nfsv3-cg%2FGUID-2111ACFD-42FB-4AE7-B5CA-896751511B40.html)

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
na_ontap_ldap_client

### CODE REVIEW

File: lib/ansible/modules/storage/netapp/na_ontap_ldap_client.py [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+    default: 'present'
+    type: str
+
+  vserver:
+    description:
+    - vserver/svm that holds LDAP client configuration
+    required: true
+    type: str
+
+  client_config:
```
Comment (lonico) [inline lib/ansible/modules/storage/netapp/na_ontap_ldap_client.py]: I would prefer to say name.  But I’m ok if you want to stick with this here.  Or maybe client_config_name.

File: lib/ansible/modules/storage/netapp/na_ontap_ldap_client.py [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+            'min_bind_level',
+            'port',
+            'query_timeout',
+            'referral_enabled',
+            'session_security',
+            'use_start_tls'
+        ]
+
+        # Handle required parameters for state=present
+        if self.module.params.get('state') == 'present':
```
Comment (lonico) [inline lib/ansible/modules/storage/netapp/na_ontap_ldap_client.py]: Ansible already has required_if.   Isn’t it possible to use it here?

File: lib/ansible/modules/storage/netapp/na_ontap_ldap_client.py [hunk 3]
```diff
@@ -0,0 +1,365 @@
...
+        '''
+        # Make query
+        client_config_info = netapp_utils.zapi.NaElement('ldap-client-get-iter')
+
+        if client_config_name is None:
+            client_config_name = self.parameters['client_config']
+
+        query_details = netapp_utils.zapi.NaElement.create_node_with_children('ldap-client', **{'ldap-client-config': client_config_name})
+
+        query = netapp_utils.zapi.NaElement('query')
```
Comment (lonico) [inline lib/ansible/modules/storage/netapp/na_ontap_ldap_client.py:218]: It could be safer to add vserver to the query

**Review by lonico — APPROVED** [review]
shipit

File: lib/ansible/modules/storage/netapp/na_ontap_ldap_client.py (hunk 3 above)
Comment (zeten30) [reply lib/ansible/modules/storage/netapp/na_ontap_ldap_client.py:218]: I've just pushed updated code. Please check it.

**Review by lonico — APPROVED** [review]
shipit

**Review by carchi8py — APPROVED** [review]
shipit

**Comment by zeten30** [comment]
Close & reopen to get rid of 'stale_ci' as suggested by ansibot :)

**Review by thedoubl3j — APPROVED** [review]
module is coming under netapp certified umbrella. approved by maintainers @lonico and @carchi8py. shipit

**Pull request merged**